test/unit/storage/AsyncLockResourceLocker.test.ts

```diff
@@ -0,0 +1,66 @@
+import { AsyncLockResourceLocker } from '../../../src/storage/AsyncLockResourceLocker';
+
+describe('An AsyncLockResourceLocker', (): void => {
+  let locker: AsyncLockResourceLocker;
+  beforeEach(async(): Promise<void> => {
+    locker = new AsyncLockResourceLocker();
+  });
+
+  it('can acquire a lock.', async(): Promise<void> => {
+    const lock = await locker.acquire({ path: 'path' });
+    expect(lock).toEqual(expect.objectContaining({ release: expect.any(Function) }));
+  });
+
+  it('can release an acquired lock.', async(): Promise<void> => {
+    const lock = await locker.acquire({ path: 'path' });
+    await expect(lock.release()).resolves.toBeUndefined();
+  });
+
+  it('can acquire a lock after it was released.', async(): Promise<void> => {
+    let lock = await locker.acquire({ path: 'path' });
+    await lock.release();
+    lock = await locker.acquire({ path: 'path' });
+    expect(lock).toEqual(expect.objectContaining({ release: expect.any(Function) }));
+  });
+
+  it('blocks lock acquisition until they are released.', async(): Promise<void> => {
```

Comment:
Is this really what we are testing?

Reply:
Haha, yes it is. But it took me some time to get it; I had missed the swapped orders of the locks. Perhaps add some comments to emphasize that?